nearcore/src/config.rs

```diff
@@ -555,10 +566,16 @@ impl NearConfig {
                 catchup_step_period: config.consensus.catchup_step_period,
                 chunk_request_retry_period: config.consensus.chunk_request_retry_period,
                 doomslug_step_period: config.consensus.doomslug_step_period,
-                tracked_accounts: config.tracked_accounts,
-                tracked_shards: config.tracked_shards,
-                tracked_shadow_validator: config.tracked_shadow_validator,
-                tracked_shard_schedule: config.tracked_shard_schedule.unwrap_or(vec![]),
+                tracked_config: if let Some(tracked_config) = config.tracked_config {
+                    tracked_config
+                } else {
+                    TrackedConfig::from_deprecated_config_values(
+                        config.tracked_shards,
+                        config.tracked_shard_schedule,
+                        config.tracked_shadow_validator,
+                        config.tracked_accounts,
+                    )
+                },
```

Comment:
nit: Can this logic be moved to a method in Config? 
Ideally we would also make relevant fields private but I'm not sure if that's possible

Reply:
marked as pub(crate) because these field are used in `config_validate.rs`